keps/sig-windows/4885-windows-cpu-and-memory-affinity/README.md

```diff
@@ -0,0 +1,594 @@
+related to the [Kubernetes Node Memory Management](#kubernetes-nodes-memory-management-mechanisms-and-their-relation-to-the-memory-manager).
+
+### Windows Topology manager considerations
+
+Topology manager is already enabled on Windows in order to support the device manager.  Enabling the CPU and Memory manager as 
+hint providers will be behind a feature flag. The CPU manager and Memory Manager can independently be enabled or disabled to support cases where the features needs to be shut off.  
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+The testing plan is to enable basic tests in [Windows testing folder](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/windows) in Alpha.  This will enable us to progress to a state we in Alpha that will allow our end users to test and give feedback in real world scenarios. 
+
+We we also work to enable e2e_node test suite to run on Windows and enable the applicable [CPU](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/container_manager_test.go)/[Memory](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/memory_manager_test.go)/[Topology](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/topology_manager_test.go) Manager tests for Beta.  The goal will be to enable as many of those tests as possible while recognizing some may not be applicable to Windows.  Where we find gaps we will fill them with Windows specific tests.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- pkg/kubelet/cm/container_manager_windows.go
+- pkg/kubelet/cm/internal_container_lifecycle_windows.go
+- pkg/kubelet/winstats/cpu_topology_test.go
+
+##### Integration tests
+
+Integration tests do not run on Windows. Functionality will be covered by unit and e2e tests.
+
+##### e2e tests
+
+-  e2e_node will need to be enabled for windows to add coverage
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial basic e2e tests in Windows e2e suite are added
+- unit tests for Windows specific components are added
+
+#### Beta
+
+- Gather feedback from developers 
+- e2e_node tests are in Testgrid and linked in KEP
+
+#### GA
+
+- 2 examples of real-world usage
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+N/A
```

Comment:
Also seems relevant, especially regarding n-3 kubelet and CRI.

```
<!--
If applicable, how will the component handle version skew with other
components? What are the guarantees? Make sure this is in the test plan.

Consider the following in developing a version skew strategy for this
enhancement:
- Does this enhancement involve coordinating behavior in the control plane and nodes?
- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
- Will any other components on the node change? For example, changes to CSI,
  CRI or CNI may require updating that component before the kubelet.
-->
```

Reply:
This feature is kubelet specific and doesn't interact with any other components such as control plane or other kublets.

Reply:
@jsturtevant doesn't it interact with CRI and propose CRI changes?

Reply:
It does interact with CRI.
I can update this portion of the KEP since James is OOO until end of enhancement freeze.

Reply:
updated - PTAL @enj / @deads2k